### PR TITLE
Lists elements that matched multiple error

### PR DESCRIPTION
**Pull request by JacobParis**
**What**:

This PR adds the found elements to the log output of the MultipleError

**Current error**
```
TestingLibraryElementError: Found multiple elements with the role "definition"
```

**New error**
```
TestingLibraryElementError: Found multiple elements with the role "definition"
    <dd
      aria-labelledby="uid-4-propertyValue"
      class="mb-0 text-size-large"
    >
      565000           
    </dd>
    <dd
      aria-labelledby="uid-4-appraisedValue"
      class="mb-0 text-size-large"
    >
      None         
    </dd>
    <dd
      aria-labelledby="uid-4-appraisedValue"
      class="mb-0 text-size-large"
    >
      2240 sqft         
    </dd>
```
**Why**:

It's not always immediately obvious which elements are conflicting, and since dom-testing-library needs to find them in order to know there are multiples it makes sense to expose that to the user

**How**:

I'm quite certain this is not the correct implementation of this feature, but I wanted to trial it out before investing too much time. If it's something deemed worthy of inclusion, I'd appreciate a little guidance on how to structure this (possibly a new wrapQuery type function) in a way that's idiomatic to the codebase

**Checklist**:

- [ ] Documentation added to the
      [docs site](https://github.com/testing-library/testing-library-docs) N/A
- [x] Tests
- [ ] Typescript definitions updated N/A
- [x] Ready to be merged

### CODE REVIEW

File: src/query-helpers.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ function makeSingleQuery(allQuery, getMultipleError) {
     const els = allQuery(container, ...args)
     if (els.length > 1) {
       throw getMultipleElementsFoundError(
-        getMultipleError(container, ...args),
+        getMultipleError({container, elements: els}, ...args),
         container,
```
Comment (kentcdodds) [inline src/query-helpers.js:50]: This is a breaking change

**Comment by kentcdodds** [comment]
I definitely feel like this would improve the error message. I'd love to find a way to do it that is backwards compatible. We can improve the API in the next major version bump.

**Comment by JacobParis** [comment]
I think ideally it belongs in the `getMultipleElementsFoundError` function, but since that requires an API change, directly in makeSingleQuery is the next best place

I've pushed that change (no longer breaking) and an accompanying test

**Comment by kentcdodds** [comment]
I think your other PR triggered merge conflicts. Could you fix those please?

**Comment by JacobParis** [comment]
Wow git really mangled those together

All resolved now 🙏

**Comment by kentcdodds** [comment]
Thanks so much for your help! I've added you as a collaborator on the project. Please make sure that you review the `other/MAINTAINING.md` and `CONTRIBUTING.md` files (specifically the bit about the commit messages and the git hooks) and familiarize yourself with the code of conduct (we're using [the contributor covenant](http://contributor-covenant.org/)). You might also want to [watch the repo](https://help.github.com/articles/watching-repositories/) to be notified when someone files an issue/PR. Please continue to make PRs as you feel the need (you can make your branches directly on the repo rather than your fork if you want). Thanks! And welcome to the team :)

**Pull request merged**